include/mxnet/base.h

```diff
@@ -316,6 +321,19 @@ inline Context Context::GPU(int32_t dev_id) {
   return Create(kGPU, dev_id);
 }
 
+inline int32_t Context::GetGPUCount() {
+#if MXNET_USE_CUDA
+  int32_t count;
+  cudaError_t e = cudaGetDeviceCount(&count);
+  if (e != cudaSuccess) {
+    return 0;
+  }
+  return count;
+#else
+  return 0;
```

Comment:
Machine has no gpu and mxnet not compiled with gpu should be treated separately. This should raise an error too.

Reply:
So you would raise an exception saying that MXNet was not compiled with CUDA support?

Reply:
yes

Reply:
use LOG(FALTA) << "xxx"